v2/goldie.go

```diff
@@ -167,6 +206,36 @@ func (g *Goldie) Update(t *testing.T, name string, actualData []byte) error {
 	return nil
 }
 
+func (g *Goldie) UpdateWithTemplate(t *testing.T, name string, data interface{}, actualData []byte) error {
```

Comment:
Add comment and expose this in the interface (or make private, I'm ok either way)